public/docs/ts/latest/cookbook/validation.jade

```diff
@@ -0,0 +1,263 @@
+
+   Using the model-driven approach to form validation, the validation rules are specified in the model as defined in the
+   component class. Defining the validation in the class instead of the template gives you more control. You
+   can adjust the validation based on the application state or user and you can unit test the validation rules.
+   Your code then becomes the source of truth for your validation.
+
+   By combining a model-driven approach with data binding, you can get the best of the model-driven approach, but
+   still have some of the simplicity of the template-driven approach.
+
+.alert.is-important
+  :marked
+    When moving the validation attributes out of the HTML, we are no longer aria ready. Work is being done to
+    address this. 
+
++makeExample('cb-validation/ts/app/hero-form-model-binding.component.ts','class','app/hero-form-model-binding.component.ts')
+
+:marked
+   In the component class, we define the form and the controls on the form:
+      - Declare a property for the form typed as a `ControlGroup`.
+      - Declare a property for each control on the form typed as `Control`. You only need to define a property for a control
+      if the control has validation rules. Notice that in our example, the alter ego is not validated so it is not declared here.
+      - Add a constructor for the class and use dependency injection to inject in the `FormBuilder` service. We use
+      that service to build the form.
+      - Build a method to create the form. We name this method `buildForm` in our example.
+      The form is created in a method instead of directly within the constructor so it can be
+      called again to reset the form and its validation when adding a new hero.
+      - In the `buildForm` method, create an instance of each control, defining the appropriate validation rules.
+        - The first argument of the `Control` constructor is the default value. We don't need to set a default here because
+      we are using data binding in the template. That data binding provides a default value as appropriate.
+        - The second argument of the `Control` constructor defines the validation rules for the control. Only one validation
+      rule can be defined here, as shown with the `powerControl` property. If there are multiple validation rules, use the
+      `Validator.compose` method to define an array of validation rules, as shown with the `nameControl` property.
+        - The optional third argument of the `Control` constructor (not shown) defines any asynchronous validation rules.
+      - Group the controls into a form group using the `FormBuilder` instance. Here we also add the `alterEgo` so the form
+      definition is complete.
+      - We call this `buildForm` method from the constructor. But in many applications, you may want to call `buildForm`
+      from somewhere else such as the `ngInit` lifecycle hook.
+
+   We'll use the class form and control properties in the template. 
```

Comment:
Which sentence are you referring to? Is there something marked in the code above?